api/include/mfxdefs.h

```diff
@@ -50,6 +50,29 @@ extern "C"
 {
 #endif /* __cplusplus */
 
+#if defined(_MSC_VER)
+    #define MFX_PRAGMA_IMPL(x) __pragma(x)
+#else
+    #define MFX_PRAGMA_IMPL(x) _Pragma(#x)
+#endif
+
+#define MFX_PACK_BEGIN_X(x) MFX_PRAGMA_IMPL(pack(push, x))
+
+#if defined(_WIN64) || defined(__LP64__)
+    #define MFX_PACK_BEGIN()     MFX_PACK_BEGIN_X(8)
+    #define MFX_PACK_BEGIN_MAX() MFX_PACK_BEGIN()
+    #define x64
+#elif defined(_WIN32) || defined(_M_IX86) && !defined(__linux__)
+    #define MFX_PACK_BEGIN()     MFX_PACK_BEGIN_X(4)
+    #define MFX_PACK_BEGIN_MAX() MFX_PACK_BEGIN_X(8)
+#elif defined(_M_IX86) && defined(__linux__) || defined(__ILP32__)
+    #define MFX_PACK_BEGIN()     MFX_PACK_BEGIN_X(4)
+    #define MFX_PACK_BEGIN_MAX() MFX_PACK_BEGIN()
```

Comment:
I don't intuitively understand when you are going to use MFX_PACK_BEGIN and MFX_PACK_BEGIN_MAX. Do names signify anything special? Why _MAX?

Reply:
Maybe there is a better naming. Meaning was to guarantee correct alignment of long long int / double which is different across 32 win/lin